command/agent/local.go

```diff
@@ -434,11 +434,28 @@ func (l *localState) setSyncState() error {
 		if l.config.CheckUpdateInterval == 0 {
 			equal = existing.IsSame(check)
 		} else {
+			// Copy the existing check before potentially modifying
+			// it before the compare operation.
 			eCopy := new(structs.HealthCheck)
 			*eCopy = *existing
-			eCopy.Output = ""
-			check.Output = ""
-			equal = eCopy.IsSame(check)
+
+			// Copy the server's check before modifying, otherwise
+			// in-memory RPC-based unit tests will have side effects.
+			cCopy := new(structs.HealthCheck)
+			*cCopy = *check
```

Comment:
Even though the Copy() method will be just an assignment, can this be moved to a `Copy()` method?  If `HealthCheck` ever picks up something that can't be shallow copied, we won't have to dig around finding all assignments?